File: Scripts/film-cs/Logger.cs
using System;

namespace FilmCS
{
    public class Logger
    {
        private readonly string _name;
        public Logger(string name)
        {
            _name = name;
        }

        public void Debug(string msg)
        {
            Console.WriteLine($"[DEBUG] [{_name}] {msg}");
        }

        public void Info(string msg)
        {
            Console.WriteLine($"[INFO] [{_name}] {msg}");
        }

        public void Warning(string msg)
        {
            Console.WriteLine($"[WARNING] [{_name}] {msg}");
        }

        public void Error(string msg)
        {
            Console.WriteLine($"[ERROR] [{_name}] {msg}");
        }

        public void Critical(string msg)
        {
            Console.WriteLine($"[CRITICAL] [{_name}] {msg}");
        }
    }
}

File: Scripts/film/logger.py


import logging


class Logger:
    def __init__(self, name: str, level: int = logging.DEBUG):
        self.logger = logging.getLogger(name)
        self.logger.setLevel(level)

        if not self.logger.hasHandlers():
            # Console handler
            console_handler = logging.StreamHandler()
            console_handler.setLevel(level)

            formatter = logging.Formatter(
                "[%(asctime)s] %(name)s %(levelname)s: %(message)s", datefmt="%H:%M:%S"
            )
            console_handler.setFormatter(formatter)

            self.logger.addHandler(console_handler)

    def debug(self, msg: str, *args, **kwargs):
        self.logger.debug(msg, *args, **kwargs)

    def info(self, msg: str, *args, **kwargs):
        self.logger.info(msg, *args, **kwargs)

    def warning(self, msg: str, *args, **kwargs):
        self.logger.warning(msg, *args, **kwargs)

    def error(self, msg: str, *args, **kwargs):
        self.logger.error(msg, *args, **kwargs)

    def critical(self, msg: str, *args, **kwargs):
        self.logger.critical(msg, *args, **kwargs)

    def set_level(self, level: int):
        self.logger.setLevel(level)

    def get_logger(self) -> logging.Logger:
        return self.logger